portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,47 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // response.setContentType("text/html;");
+    // response.getWriter().println("<h1>Hello Sofia</h1>");
+    ArrayList<String> array = new ArrayList<String>();
+    array.add("Bogota");
+    array.add("Miami");
+    array.add("Atlanta");
+    String json = convertToJson(array);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList array) {
+      String json = "{";
+      json += "\"text1\": ";
+      json += "\"" + array.get(0) + "\"";
+      json += ", ";
+      json += "\"text2\": ";
+      json += "\"" + array.get(1) + "\"";
+      json += ", ";
+      json += "\"text3\": ";
+      json += "\"" + array.get(2) + "\""; 
+      json += "}";
+
+      return json;
```

Comment:
When you have this many fragments to add together, it is more efficient runtime-wise to use StringBuilder than default concatenation.

https://docs.oracle.com/javase/tutorial/java/data/buffers.html

Reply:
oooo ok that makes sense since String cat in Java is slow.